src/System.Management.Automation/engine/AutomationEngine.cs

```diff
@@ -38,27 +38,19 @@ internal AutomationEngine(PSHost hostInterface, InitialSessionState iss)
 #if !UNIX
             // Update the env variable PathEXT to contain .CPL
             var pathext = Environment.GetEnvironmentVariable("PathEXT");
-            pathext = pathext ?? string.Empty;
-            bool cplExist = false;
-            if (pathext != string.Empty)
+
+            if (string.IsNullOrEmpty(pathext))
             {
-                string[] entries = pathext.Split(Utils.Separators.Semicolon);
-                foreach (string entry in entries)
-                {
-                    string ext = entry.Trim();
-                    if (ext.Equals(".CPL", StringComparison.OrdinalIgnoreCase))
-                    {
-                        cplExist = true;
-                        break;
-                    }
-                }
+                Environment.SetEnvironmentVariable("PathEXT", ".CPL");
             }
-
-            if (!cplExist)
+            else if (!(pathext.EndsWith(";.CPL", StringComparison.OrdinalIgnoreCase) ||
+                       pathext.StartsWith(".CPL;", StringComparison.OrdinalIgnoreCase) ||
+                       pathext.Contains(";.CPL;", StringComparison.OrdinalIgnoreCase) ||
+                       pathext.Equals(".CPL", StringComparison.OrdinalIgnoreCase)))
             {
-                pathext = (pathext == string.Empty) ? ".CPL" :
-                    pathext.EndsWith(";", StringComparison.OrdinalIgnoreCase)
-                    ? (pathext + ".CPL") : (pathext + ";.CPL");
+                // Fast skip if we already added the extention as ";.CPL".
+                // Fast skip if user already added the extention.
+                pathext = pathext + ";.CPL";
                 Environment.SetEnvironmentVariable("PathEXT", pathext);
```

Comment:
Can you use `PATHEXT` instead? That's the default name.

Reply:
Done.